src/test/ReactTestUtils.js

```diff
@@ -61,12 +67,38 @@ function findAllInRenderedTreeInternal(inst, test) {
     typeof currentElement.type === 'function'
   ) {
     ret = ret.concat(
-      findAllInRenderedTreeInternal(inst._renderedComponent, test)
+      findAllInRenderedStackTreeInternal(inst._renderedComponent, test)
     );
   }
   return ret;
 }
 
+function findAllInRenderedFiberTreeInternal(fiber, test) {
+  if (!fiber) {
+    return [];
+  }
+  if (
+    fiber.tag !== ClassComponent &&
+    fiber.tag !== HostComponent &&
+    fiber.tag !== HostText
+  ) {
+    return [];
+  }
+  var publicInst = fiber.stateNode;
+  var ret = publicInst && test(publicInst) ? [publicInst] : [];
+  var child = fiber.child;
+  while (child) {
```

Comment:
Can you add that comment about coroutines that input everywhere? Just so I can find it later. Coroutines have children on stateNode atm.